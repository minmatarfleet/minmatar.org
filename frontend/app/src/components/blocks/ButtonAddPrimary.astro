---
import { i18n } from '@helpers/i18n'
const { translatePath } = i18n(Astro.url)

import type { ButtonColors, ButtonSizes } from '@dtypes/layout_components';

interface Props {
    color?:         ButtonColors;
    size?:          ButtonSizes;
    narrow?:        boolean;
    dialog_title:   string;
    redirect?:      string;
}

const {
    color = 'fleet-red',
    size = 'lg',
    narrow = false,
    dialog_title,
    redirect = Astro.url,
} = Astro.props;

import { query_string } from '@helpers/string';
const ADD_PRIMARY_CHARACTER_INIT_PARTIAL_URL = `${translatePath('/redirects/add_primary_character_init')}?redirect=${redirect}`

import Button from '@components/blocks/Button.astro';
---

<Button
    type="button"
    color={color}
    size={size}
    narrow={narrow}
    x-data={`{
        add_primary_pilot(accepted) {
            if (accepted) window.location.href = "${ADD_PRIMARY_CHARACTER_INIT_PARTIAL_URL}"
        },
        show_add_pilot_disclaimer() {
            show_alert_dialog({
                title: '${dialog_title}',
                partial: '${translatePath('/partials/add_primary_pilot_dialog_disclaimer/')}?${query_string({
                    is_main: JSON.stringify(true),
                })}',
            }).then( (accepted) => this.add_primary_pilot(accepted) )
        }
    }`}
    x-on:click="show_add_pilot_disclaimer"
>
    <slot />
</Button>