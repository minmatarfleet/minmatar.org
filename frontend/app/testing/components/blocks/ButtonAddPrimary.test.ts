import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ButtonAddPrimary from "@components/blocks/ButtonAddPrimary.astro";

test("ButtonAddPrimary defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ButtonAddPrimary, {});

  expect(result).toMatchSnapshot();
});
