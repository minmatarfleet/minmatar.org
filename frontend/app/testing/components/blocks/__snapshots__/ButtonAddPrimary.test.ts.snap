// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonAddPrimary defaults 1`] = `
"<button type="button" x-data="{
        add_primary_pilot(accepted) {
            if (accepted) window.location.href = &#34;/redirects/add_primary_character_init?redirect=https://example.com/&#34;
        },
        show_add_pilot_disclaimer() {
            show_alert_dialog({
                title: 'undefined',
                partial: '/partials/add_primary_pilot_dialog_disclaimer/?is_main=true',
            }).then( (accepted) => this.add_primary_pilot(accepted) )
        }
    }" x-on:click="show_add_pilot_disclaimer" data-size="lg" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">       </div>   </button> "
`;
